e2e/tests/kube_client.go

```diff
@@ -0,0 +1,43 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"os/exec"
+)
+
+type CommandResult struct {
+	Stdout string
+	Stderr string
+	Err    error
+}
+
+var _ PocketClient = &KubeClient{}
+
+// PocketClient is a single function interface for interacting with a node.
+type PocketClient interface {
+	RunCommand(...string) (*CommandResult, error)
+}
+
+// KubeClient saves a reference to a command
+type KubeClient struct {
```

Comment:
Ditto, whenever I see things publically exposed (capitalized), I put myself in the shoes of a new developer in the codebase and try to understand: "how should I be using this?"

Downscope until necessary. Given that it implements the `PocketClient` interface, I don't think that'll be needed.